src/averageAges.js

```diff
@@ -55,7 +85,19 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const children = !onlyWithSon
+    ? people.filter(child =>
+      (people.find(person => child.mother === person.name))
+    )
+    : people.filter(child =>
+      (people.find(person => child.mother === person.name && child.sex === 'm'))
+    );
```

Comment:
you basically do the same in these 2 filters, better move your ternary inside
```suggestion
  const children = people.filter(child => onlyWithSon
    ? (people.find(person => child.mother === person.name  && child.sex === 'm'))
    : (people.find(person => child.mother === person.name)
  )
```